### DISCUSSION TITLE
English sentences that should be supported

### DISCUSSION THREAD

**Discussion by balloob** [OHF-Voice/intents#422] [Supported Sentences]
If you don't feel comfortable contributing to the YAML in this repository, you can post your **English** sentences that match [our supported intents](https://github.com/home-assistant/intents/blob/main/intents.yaml) below.

**Comment by alpharesearch** [comment]
I have absolutely no idea what your guy are collecting here. In the end it would be nice if this could replace my google home with smartthings setup that I have in parallel to HA. I don’t use the search feature or play music feature all that often mainly turning stuff on and off like switches, lights, automations, rooms. I do use the set a timer all the time, really helpful when cooking or making tea. I normally just say either turn/switch on or turn/switch off or maybe activate or deactivate for routine/scene/automation. I guess there are other words like flip that could be use, but I personally don't use that. The entity could be before or after the on/off/activate/deactivate. Some entity names may have light or lights or switch in the name. hopefully this will not confuse. Here are some examples I use:
Turn the kitchen light on.
Switch off the kitchen light.
I have four lights in my bedroom. Saying "turn the bedroom lights off", should turn all four lights off.
Other commands I use are:
Set a timer for 5 minutes.
Set the thermostat to 62.
"Good Night", this locks all doors, turns a list of lights off, turns all TVs off, turns all music that is playing off, lays bed down, etc.

**Reply by synesthesiam** [reply]
We are collecting common ways of saying certain HA commands, like some of your examples. These are written as [templates](https://github.com/home-assistant/intents/blob/main/docs/templates.md) so you can mark certain words as optional, etc.

The goal here is to support a basic set of commands, but the system in HA can be easily extended with custom commands.

**Reply by alpharesearch** [reply]
Thanks for the explanation.   
One other thing to consider is to make the spelling more fuzzy. For example, my name is Markus with a k and I have a Markus Lamp, but when I speak the name to home assistant,  the voice recognition spells it Marcus with a c, and my light doesn't switch.

**Reply by schizza** [reply]
@alpharesearch you can make an alias for that light with name Marcus :)

**Reply by alpharesearch** [reply]
> @alpharesearch you can make an alias for that light with name Marcus :)

Good workaround, but I don't have to do this on Google home.

**Comment by Martin-Dolan** [comment]
not 100% sure on how to contribute and don't want to screw things up so hope this helps...
I have some gates that I get Home assistant /google home to control

Need to add "gate"  

examples:
open the gate
open the [front/side/back/rear/pedestrian] gate

close/shut the gate
close/shut the [front/side/back/rear/pedestrian] gate

is the [front/side/back/rear/pedestrian] gate  open/closed/shut?

**Comment by heytcass** [comment]
Every morning my wife turns all the lights in the house with "Set all the lights to 100%." I just tried that with Assistant, and got "Sorry, I didn't understand that"

**Reply by synesthesiam** [reply]
Our sentences are targeting individual entities and entities in a specific area. We're not sure yet about sentences that affect everything.

But, this is something that could be done pretty easily in a custom sentence!

**Reply by OzGav** [reply]
@heytcass @synesthesiam I've just upvoted this. Additionally, my suggestion is to have the term "in the house" as an optional at the end of the sentence as well.  For example, `What lights are on in the house` would be the same as `What lights are on`

Certainly every night in our house we say to Google "Turn off all the lights" as we go to bed so being able to support commands that target all areas simultaneously would be advantageous.

I will try a custom sentence to do this as suggested as I am yet to play with that!

**Comment by Bodge-IT** [comment]
OK, so I've only tried one sentence so far and that failed:
Dim {Lamp} brightness by 30%

Lamp being an actual device name.  Was surprised it din't work.

I'll come back with more as I find them  Is this how you want the info?

**Comment by titchy** [comment]
A sentence that should work but does not is this one:
"set hall temperature to 16 degrees"

**Comment by mindsocket** [comment]
I'm not sure where to post this, so trying here. I was hoping to call a service (that calls an API) with a free text value, via voice. I realised that there's no way to define an intent like "Remind me about {thing}". If I'm not mistaken {thing} can't be an arbitrary phrase, it has to be a specific known kind of "slot" with specific values. Is that right? If so, I'm curious if that's ever likely to change. I realise quite a lot of HA's sphere of interest is a known set of things with known values, but not everything is (input_text , calendar items, etc)

**Reply by tetele** [reply]
First of all, this is a very good place to discuss this matter.

Second of all, arbitrary input [will eventually be available](https://discord.com/channels/330944238910963714/1054846248374505572/1071937961685418094), having a shopping list use case in mind, but not at the moment. For the moment you can have text slots or numeric ranges.

If you want to do it right now, you can use this custom integration https://github.com/siemon-geeroms/yarvis but in a separate pipeline.

**Reply by mindsocket** [reply]
Update: after much digging around I found a similarly requested capability in Hassil [here](https://github.com/home-assistant/hassil/issues/47)

It's not resolved but I see a positive intent to do so and a custom agent workaround

**Comment by n8detar** [comment]
It would be nice if you could be able to ask where a person is located (person domain).

`where is John (located | at)`
`(what | whats | what's) [is] John['s | s] location`

Possible responses:
John is home
John is away
John is at `Zone Name` (i.e. John is at `Costco`)

**Reply by OzGav** [reply]
You could do that with a custom intent
